datafusion/spark/src/function/array/shuffle.rs

```diff
@@ -47,7 +49,7 @@ impl Default for SparkShuffle {
 impl SparkShuffle {
     pub fn new() -> Self {
         Self {
-            signature: Signature::arrays(1, None, Volatility::Volatile),
+            signature: Signature::user_defined(Volatility::Volatile),
```

Comment:
We should avoid using user_defined and make use of `TypeSignature::ArraySignature`, for example:

https://github.com/apache/datafusion/blob/7591919be7e6582ed7f6a8d0b033f7a0a8ad60f7/datafusion/expr-common/src/signature.rs#L1247-L1262

- Index can be used as seed since it is Int64 type
- We can't use `Signature::array_and_index` directly since that would coerce FixedSizeLists to List arrays, which we don't want since we have a native implementation for FixedSizeLists